### ISSUE TITLE
YDB Node.js SDK v3

### ISSUE THREAD

**Issue by rrewll** [ydb-platform/ydb-nodejs-sdk#136]
**Changes**:
- Replace `grpc` with `@grpc/js`

- Driver constructor reorganization: using IDriverSettings dynamic parameter instead of tuple of parameters.
  Before: `new Driver(entryPoint, dbName, authService, ...)`
  After: `new Driver({connectionString: "...", authService: ..., ...})` or `new Driver({endpoint: "...", database: "...", authService: "...", ...})`

- Simplifcation of authentication support:
  - `database` and `rootCertificates` are no longer needed in *AuthService constructors (rootCertificates is now initialized in the driver if it's neccessary)
     Before: `const authService = new MetadataAuthService(dbName, makeSslCredentials());`
     After: `const authService = new MetadataAuthService();`

  - `getCredentialsFromEnv()` is simplifed:
    Before: `const authService = getCredentialsFromEnv(endpoint, database, logger);`
    After: `const authService = getCredentialsFromEnv();` (logger is optional)

  - Old credential environment variables are no longer supported
  .
- Types:
   - Fix existing (tz datetime types, uuid) and support new types (decimal, container types)
   - `TypedData.createNativeObjects()` now supports non-primitive values (decimal, optional, container types) in incoming result sets
   - Introduce Types helper class to construct primitive and container types
      Before: `@declareType({optionalType: {item: {typeId: Ydb.Type.PrimitiveTypeId.UINT64}}})`
      After: `@declareType(Types.optional(Types.UINT64))`
      Before: `new Column('series_id', Ydb.Type.create({optionalType: {item: {typeId: Ydb.Type.PrimitiveTypeId.UINT64}}}))`
      After: `new Column('series_id', Types.optional(Types.UINT64))`

   - Introduce TypedValues helper class (instead of Primitive) to construct query parameters and key ranges (in readTable requests) without TypedData
     Before: `{'$param': typedObj.getTypedValue('param')}`
     After: `{'$param': TypedValues.utf8(param)}`

- Table and Scheme clients: unify signature of client methods and introduce optional *Settings classes to avoid collisions in case of new optional parameters and simplify the usage.
   Before: `session.executeQuery(query, {}, TX_AUTO, undefined, undefined, QueryStatsCollection.Mode.FULL)`
   After: `session.executeQuery(query, {}, TX_AUTO, new ExecuteQuerySettings.withCollectStats(QueryStatsCollection.Mode.FULL))`

- Fix bug when bulkUpsert and readTable methods require a table path with database prefix
   Before: `session.readTable(database + '/' + tablePath)`
   After: `session.readTable(tablePath)`

- Refactoring and renaming:
   - rename entryPoint -> endpoint, dbName -> database in SDK code end examples
   - most of all interfaces have "I" prefix
   - Example npm commands moved from main package.json to `examples` + fix snippets in auth examples.
   - Reduce the number of exports from internal implementation
